### ISSUE TITLE
How to monitor the state of the node for testing

### ISSUE THREAD

**Issue by hackaugusto** [0xPolygonMiden/miden-node#97]
- Decide on what to monitor
- Then how to monitor

related: https://github.com/0xPolygonMiden/miden-node/issues/95 https://github.com/0xPolygonMiden/miden-node/issues/82

**Comment by hackaugusto** [comment]
This is related to issue #82 , but the goal is not production, but testing.

**Comment by bobbinth** [comment]
One set of useful endpoints specifically for the store could be:
1. List all accounts and their current state. This would be very simple - for test deployments only.
2. List all notes with their metadata (in descending order) - similar to the above.
3. List all nullifiers - also similar to the above (though, maybe we don't need it).

We could use these manually to check the state of the store (e.g., after we execute a transaction).

**Comment by bobbinth** [comment]
Another useful thing to add would be some kind of "heartbeat" endpoint on the RPC component. This could return the number of the latest block in the chain.

**Comment by bobbinth** [comment]
Regarding "how to monitor", I've tried using [Postman](https://www.postman.com/downloads/) and it works quite well.

All I had to do was the following:

1. Download Postman.
2. Create a new request and specify "gRPC" as the type.
3. Import our `proto` files.
4. Specify server address - e.g., for the RPC component it is `grpc://localhost:57291`.

Then, I can invoke specific endpoints. For example, for `GetBlockHeaderByNumber`, I get the following response:

![Screenshot 2024-01-03 at 8 22 56 PM](https://github.com/0xPolygonMiden/miden-node/assets/43513081/8a5f436e-327a-4faf-8818-b280ae022c6e)
